app/src/main/kotlin/at/bitfire/davdroid/webdav/DavDocumentsProvider.kt

```diff
@@ -287,7 +287,21 @@ class DavDocumentsProvider(
             runningQueryChildren.remove(parentId)
 
         // Regardless of whether the worker is done, return the children we already have
-        for (child in documentDao.getChildren(parentId)) {
+        val children = if (sortOrder == null) {
+            documentDao.getChildren(parentId)
+        } else {
+            documentDao.getChildrenOrdered(
+                parentId,
+                // Convert the cursor's column name into Room's
+                sortOrder
+                    .replace(Document.COLUMN_DOCUMENT_ID, "id")
+                    .replace(Document.COLUMN_DISPLAY_NAME, "displayName")
+                    .replace(Document.COLUMN_MIME_TYPE, "mimeType")
+                    .replace(Document.COLUMN_SIZE, "size")
+                    .replace(Document.COLUMN_LAST_MODIFIED, "lastModified")
```

Comment:
I'd prefer a proper `mapOf()` whose entries are then all applied.

Can we safely just replace the names, i.e. is it guaranteed that Room escapes the arguments of ORDER BY?

What happens when a column doesn't exist in the database? For instance a caller may request sorting by `COLUMN_SUMMARY`. I think in that case it would be better to return the values in arbitrary order than to throw an exception.

Originally I had thought that we need to tokenize/parse the sortBy string and then replace/omit value by value. Don't know whether it's necessary. But in any case the `sortOrder` shouldn't cause queries to fail.